### PR TITLE
osd: add created_at meta

### PR DESCRIPTION
**Pull request by alram**
Signed-off-by: Alex Marangone <amarangone@digitalocean.com>

Not sure if/how useful this is to the overall community.

Internally we rely on OSD creation time to get some statistics, identify potentially bad drives or sometimes establish patterns when looking at slower OSDs.
We used to stat `<osd_dir>/whoami` but ever since the switch to tmpfs mount this has become inaccurate.

Since the stat is now in the bdev label we also add it to `ceph osd metadata`. However this means that most stats will be empty for a lot of users until the OSDs are recreated. I'm not sure how confusing this could get so I'll leave up to you to tell me what to do with that bit if you're interested in that meta addition. 

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by pluser** [comment]
@alram 

> Not sure if/how useful this is to the overall community.

This new metadata looks good.

BTW, there is another issue that is trying to embed Ceph version to bluestore at the OSD creation time to find whether a osd might hit a ceph version specific bug or not.

https://tracker.ceph.com/issues/57515

How about implementing this metadata ("created_ceph_version") in this PR too?
If you'd like to focus on just one metadata in this PR, I'll send a separate PR.

**Comment by alram** [comment]
Funny coincidence we talked internally about doing the exact same thing. I'll can add it to this PR np :)

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -2130,6 +2130,10 @@ int OSD::write_meta(CephContext *cct, ObjectStore *store, uuid_d& cluster_fsid,
       return r;
   }
 
+  r = store->write_meta("created_ceph_version", pretty_version_to_str());
```
Comment (ifed01) [inline src/osd/OSD.cc:2133]: Have you intentionally omitted this field retrieval in _collect_metadata? If so - why?

Comment (ifed01) [reply src/osd/OSD.cc:2133]: ah... seen the comment above... IMO it makes sense to have this field in "ceph osd metadata" output as well.

Comment (ifed01) [reply src/osd/OSD.cc:2133]: but may be rename to "ceph_version_when_created" ?

Comment (alram) [reply src/osd/OSD.cc]: Renamed and added to `_collect_metadata`. 

Should output:
```
# cat /var/lib/ceph/osd/ceph-32/{ceph_version_when_created,created_at}
ceph version 16.2.7-1084-g3b0c2317b26 (3b0c2317b265e917b7a79d87f5b5a9a8c770bbab) pacific (stable)
2022-10-05T15:18:42.962271Z

# ceph osd metadata 32| jq -r '[ .created_at, .ceph_version_when_created]'
[
  "2022-10-05T15:18:42.962271Z",
  "ceph version 16.2.7-1084-g3b0c2317b26 (3b0c2317b265e917b7a79d87f5b5a9a8c770bbab) pacific (stable)"
]
```

**Comment by ljflores** [comment]
Rados suite review: http://pulpito.front.sepia.ceph.com/?branch=wip-yuri7-testing-2022-10-17-0814

Failures:, unrelated
    1. https://tracker.ceph.com/issues/57311
    2. https://tracker.ceph.com/issues/52321
    3. https://tracker.ceph.com/issues/52657
    4. https://tracker.ceph.com/issues/57935

Details:
    1. rook: ensure CRDs are installed first - Ceph - Orchestrator
    2. qa/tasks/rook times out: 'check osd count' reached maximum tries (90) after waiting for 900 seconds - Ceph - Orchestrator
    2. rados/thrash-erasure-code: wait_for_recovery timeout due to "active+clean+remapped+laggy" pgs - Ceph - RADOS
    3. MOSDPGLog::encode_payload(uint64_t): Assertion `HAVE_FEATURE(features, SERVER_NAUTILUS)' - Ceph - RADOS
    4. all test jobs get stuck at "Running task ansible.cephlab..." - Infrstructure - Sepia

**Pull request merged**